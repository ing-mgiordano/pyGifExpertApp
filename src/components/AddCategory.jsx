import { useState } from 'react'

const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = (e) => {
        setInputValue(e.target.value)
    }
    const handleSubmit = (e) => {
        e.preventDefault()
        if(inputValue.trim().length <= 1) return
        onNewCategory( inputValue.trim() )
        setInputValue('')
    }

  return (
    <form onSubmit={ handleSubmit }>
        <input
            type="text"
            placeholder="Buscar gif"
            value={ inputValue }
            onChange={ onInputChange }  // esto { (e) => onInputChange(e) } se puede resumir a { onInputChange }
        />
    </form>
  )
}

export default AddCategory