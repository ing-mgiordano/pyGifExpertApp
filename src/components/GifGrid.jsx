import { getGif } from "../helpers/getGifs"

const GifGrid = ({ category }) => {


    getGif(category)


    return (
        <>
            <h3>{ category }</h3>
            <p>hola mundo</p>
        </>
    )
}

export default GifGrid