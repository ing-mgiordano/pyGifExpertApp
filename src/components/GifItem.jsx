import React from 'react';
import propTypes from "prop-types"

export const GifItem = ( {title, url} ) => {
  return (
    <div className="card">
      <img src={ url } alt={ title } />
      <p>{ title }</p>
    </div>
  )
}

GifItem.propTypes = {
  title: propTypes.string.isRequired,
  url: propTypes.string.isRequired,
}



// agregar prop-types => yarn add prop-types