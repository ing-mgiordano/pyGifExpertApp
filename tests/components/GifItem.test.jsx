import React from 'react';
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Prueba en <GifItem />', () =>{

    const title = 'Saitana'
    const url = 'https://one-punch.com/saitana.jpg'

    test('debe de hacer match con el snapshot ', () => {

        const { container } = render( <GifItem title={ title } url={ url } />)
        expect( container ).toMatchSnapshot()

    });

    test('debe mostrar la imagen con el URL y ALT inicado', () => {

        render( <GifItem title={ title } url={ url } />)
        //screen.debug() es para mostrar como es el componente

        //expect( screen.getByRole('img').src ).toBe( url )
        //expect( screen.getByRole('img').alt ).toBe( title )
        // si desestructuro screen.getByRole hago un codigo mas limpio
        const { src, alt } = screen.getByRole('img')
        expect( src ).toBe( url ) //deben ser iguales
        expect( alt ).toBe( title )

    });

    test('debe de mostrar el titulo del componente', () => {

        render( <GifItem title={ title } url={ url } />)
        expect(screen.getByText( title )).toBeTruthy() //debe existir el titulo
        
    });
})