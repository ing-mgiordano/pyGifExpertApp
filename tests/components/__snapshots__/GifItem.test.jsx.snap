// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem /> debe de hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitana"
      src="https://one-punch.com/saitana.jpg"
    />
    <p>
      Saitana
    </p>
  </div>
</div>
`;
